File: app/src/main/java/com/webviewblock/app/WebviewBlockSharedPreferences.java
package com.webviewblock.app;

import android.content.SharedPreferences;
import javax.inject.Singleton;

@Singleton
public class WebviewBlockSharedPreferences implements SharedPreferenceProvider {

    SharedPreferences sharedPreferences;
    public WebviewBlockSharedPreferences(SharedPreferences sharedPreferences) {
        this.sharedPreferences = sharedPreferences;
    }

    @Override
    public String getString(String key, String defValue) {
        return sharedPreferences.getString(key, defValue);
    }

    @Override
    public void putString(String key, String value) {
        sharedPreferences.edit().putString(key, value).apply();
    }

    @Override
    public boolean getBoolean(String key, Boolean defValue) {
        return sharedPreferences.getBoolean(key, defValue);
    }

    @Override
    public void putBoolean(String key, Boolean value) {
        sharedPreferences.edit().putBoolean(key, value).apply();
    }

    @Override
    public int getInt(String key, Integer defValue) {
        return sharedPreferences.getInt(key, defValue);
    }

    @Override
    public void putInt(String key, Integer value) {
        sharedPreferences.edit().putInt(key, value).apply();
    }

    @Override
    public void putLong(String key, Long value) {
        sharedPreferences.edit().putLong(key, value).apply();
    }

    @Override
    public long getLong(String key, Long defValue) {
        return sharedPreferences.getLong(key, defValue);
    }

    @Override
    public void clearAll() {
        sharedPreferences.edit().clear().apply();
        sharedPreferences.edit().commit();
    }
}

File: app/src/main/java/com/webviewblock/app/WebviewBlockSharedPreferences.kt
package com.webviewblock.app

import android.content.SharedPreferences
import javax.inject.Singleton

@Singleton
class WebviewBlockSharedPreferences(val sharedPreferences: SharedPreferences) : SharedPreferenceProvider {
    override fun getBoolean(key: String, defValue: Boolean): Boolean {
        return sharedPreferences.getBoolean(key, defValue)
    }

    override fun putBoolean(key: String, value: Boolean) {
        sharedPreferences.edit().putBoolean(key, value).apply()
    }

    override fun getInt(key: String, defValue: Int): Int {
        return sharedPreferences.getInt(key, defValue)
    }

    override fun putInt(key: String, value: Int) {
        sharedPreferences.edit().putInt(key, value).apply()
    }

    override fun getString(key: String, defValue: String): String {
        return sharedPreferences.getString(key, defValue) ?: defValue
    }

    override fun putString(key: String, value: String) {
        sharedPreferences.edit().putString(key, value).apply()
    }

    override fun getLong(key: String, defValue: Long): Long {
        return sharedPreferences.getLong(key, defValue)
    }

    override fun putLong(key: String, value: Long) {
        sharedPreferences.edit().putLong(key, value).apply()
    }

    override fun clearAll() {
        sharedPreferences.edit().clear().apply()
        sharedPreferences.edit().commit()
    }

}